tests/functional/demo/test_demo.py

```diff
@@ -241,6 +241,36 @@ def test_stress_unique_folder_creation(database, monkeypatch, tmpdir, capfd):
     capfd.readouterr()  # Suppress fd level 1 & 2
 
 
+@pytest.mark.usefixtures("clean_db")
+@pytest.mark.usefixtures("null_db_instances")
+def test_working_dir_argument_cmdline(database, monkeypatch, tmp_path):
+    """Check that a permanent directory is used instead of tmpdir"""
+    monkeypatch.chdir(os.path.dirname(os.path.abspath(__file__)))
+    orion.core.cli.main(["hunt", "-n", "allo", "--working-dir", str(tmp_path / "test"),
+                         "--max-trials", "2", "--config", "./database_config.yaml",
+                         "./black_box.py", "-x~uniform(-50,50)"])
+
+    exp = list(database.experiments.find({'name': 'allo'}))[0]
+    assert exp['working_dir'] == str(tmp_path / "test")
```

Comment:
You should test that the folder does not exist before the call to `main` and delete it at the end of the unit-test.